myfitnesspal/client.py

```diff
@@ -804,3 +810,535 @@ def get_food_item_details(self, mfp_id: int) -> FoodItem:
+
+        # Get Request for old goal values
+        old_goals_url = parse.urljoin(
+            self.BASE_API_URL, f"v2/nutrient-goals?date={today}"
+        )
+        old_goals_document = self.session.get(old_goals_url, headers=auth_header)
+        old_goals = json.loads(old_goals_document.text)
+
+        # Marcro Calculation
+        # If no macro goals were provided calculate them with percentage value
+        if carbohydrates is None or protein is None or fat is None:
+            # If even no macro percentages values were provided calculate them from old values
+            if (
+                percent_carbohydrates is None
+                or percent_protein is None
+                or percent_fat is None
+            ):
+                if energy is None:
+                    logger.warning("No energy value and no macro values provided! - Not able to update goals without a mandotory amount of information.")
+                    return False
+                else:
+                    old_energy_value = old_goals["items"][0]["default_goal"]["energy"][
+                        "value"
+                    ]
+                    old_energy_unit = old_goals["items"][0]["default_goal"]["energy"][
+                        "unit"
+                    ]
+                    old_carbohydrates = old_goals["items"][0]["default_goal"][
+                        "carbohydrates"
+                    ]
+                    old_fat = old_goals["items"][0]["default_goal"]["fat"]
+                    old_protein = old_goals["items"][0]["default_goal"]["protein"]
+
+                    # If old and new values are in diffrent units then convert old value to new unit
+                    if not old_energy_unit == energy_unit:
+                        if old_energy_unit == "kilojoules" and energy_unit == "calories":
+                            old_energy_value *= 0.2388
+                            old_energy_unit = "calories"
+                        elif old_energy_unit == "calories" and energy_unit == "kilojoules":
+                            """FROM MFP JS
+                            if (energyUnit === 'kilojoules') {
+                                calories *= 4.184;
+                            }
+                            """
+                            old_energy_value *= 4.184
+                            old_energy_unit = "kilojoules"
+                        else:
+                            raise ValueError
+
+                    carbohydrates = energy * old_carbohydrates / old_energy_value
+                    protein = energy * old_protein / old_energy_value
+                    fat = energy * old_fat / old_energy_value
+            # If percentage values were provided check
+            else:
+                if energy is None:
+                    logger.warning("No energy value and no macro values provided! - Not able to update goals without a mandotory amount of information.")
```

Comment:
This is also a situation in which we should be raising an exception instead of just logging.
```suggestion
                    raise ValueError("No energy value and no macro values provided! - Not able to update goals without a mandotory amount of information.")
```